algorithms/HPdclassifier/R/hpdegbm.R

```diff
@@ -95,240 +82,292 @@ hpdegbm <- function(
-        stop("'nExecutor' must be a positive integer number")
-
-   if(missing(distribution))
-	stop("'distribution' is a required argument")
-
-   if ( (is.na(distribution)) && (!(distribution=="gaussian")) && (!(distribution=="bernoulli")) && (!(distribution=="adaboost")) && (!(distribution=="multinomial"))  )
-       stop("'distribution' must be gaussian or bernoulli or adaboost or multinomial")
-
-  if (!( (is.numeric(n.trees)) && (length(n.trees)==1) && (n.trees%%1 == 0) && (n.trees > 0) )) 
-        stop("'n.trees' must be a positive integer")
-
-   if (!( (is.numeric(interaction.depth)) && (length(interaction.depth)==1) && (interaction.depth%%1 == 0) && (interaction.depth > 0) )) 
-        stop("'interaction.depth' must be a positive integer")
-
-   if (!( (is.numeric(n.minobsinnode)) && (length(n.minobsinnode)==1) && (n.minobsinnode%%1 == 0) && (n.minobsinnode > 0) )) 
-        stop("'n.minobsinnode' must be a positive integer")
-
-   if ( !( (is.numeric(shrinkage)) && (length(shrinkage)==1) && (shrinkage >= 0.001) && (shrinkage <= 1) ))
-        stop("'shrinkage' must be a number in the interval [0.001,1]")
-
-   if (!( (is.numeric(bag.fraction)) && (length(bag.fraction)==1) && (bag.fraction <= 1) && (bag.fraction > 0) )) 
-        stop("'bag.fraction' must be a number in the interval (0,1]")
-
-   if(trace) {
-        cat("Start model training\n")
-        starttime <- Sys.time()
+  ##########################################################################
+  # Argument checks
+  ##########################################################################
+  if(missing(X_train))
+    stop("'X_train' is a required argument")
+
+  if(missing(Y_train))
+    stop("'Y_train' is a required argument")
+
+  if(!is.dframe(X_train) & !is.data.frame(X_train) & 
+     !is.darray(X_train) & !is.matrix(X_train)) {
+    stop("'X_train' must be a dframe or data.frame or darray or matrix")
+  } 
+
+  if (nrow(X_train) == 0) {
+    stop("'X_train' must be non-empty")
+  }
+
+  if(!(.isdarrayorframe(Y_train) || is.data.frame(Y_train) ||
+       is.matrix(Y_train) || is.vector(Y_train)) || 
+     (is.vector(Y_train) && is.na(Y_train))) 
+    stop("'Y_train' must be a dframe or data.frame or darray or matrix or numeric vector")
+
+  if (is.vector(Y_train) && length(Y_train) == 0)
+    stop("'Y_train' must be non-empty")
+
+  if (!is.vector(Y_train) && nrow(Y_train) == 0)
+    stop("'Y_train' must be non-empty")
+
+  if (is.vector(Y_train) && nrow(X_train) != length(Y_train)) 
+    stop("'Y_train' must have the same number of rows as 'X_train'")
+
+  if (!is.vector(Y_train) && nrow(X_train) != nrow(Y_train))
+    stop("'Y_train' must have the same number of rows as 'X_train'")
+
+  if ((.isdarrayorframe(X_train) && !.isdarrayorframe(Y_train)) ||
+      (.isdarrayorframe(Y_train) && !.isdarrayorframe(X_train)))
+    stop("Either both 'X_train' and 'Y_train' must be dobjects, or neither can be dobjects")
```

Comment:
I think when both 'X_train' and 'Y_train' are dobjects, they must have the same partitioning schema; right? Have you checked that?